Sources/SwiftProtobuf/Message.swift

```diff
@@ -89,21 +125,28 @@ public protocol Message: CustomDebugStringConvertible {
   // google.protobuf.Any support
   //
 
-  // Decode from an `Any` (which might itself have been decoded from JSON,
-  // protobuf, or another `Any`).
+  /// Decode from an instance of `Any` (which might itself have
+  /// been decoded from JSON, protobuf, or another `Any`).
+  ///
+  /// - Parameter any: item to decode
+  /// - Throws: an instance of `DecodingError` if the message cannot be parsed
   init(any: Google_Protobuf_Any) throws
 
   /// Serialize as an `Any` object in JSON format.
   ///
   /// For generated message types, this generates the same JSON object as
   /// `serializeJSON()` except it adds an additional `@type` field.
+  ///
+  /// - Throws: an instance of `EncodingError` on failure
   func serializeAnyJSON() throws -> String
 
   //
   // JSON encoding/decoding support
   //
 
-  /// Overridden by well-known-types with custom JSON requirements.
+  /// Serialize using the standard protobuf JSON representation.
+  ///
+  /// - Throws: an instance of `EncodingError` on failure
   func serializeJSON() throws -> String
   init(decoder: inout JSONDecoder) throws
```

Comment:
We should separately document the `init(decoder: inout JSONDecoder)` here.  It is an internal helper that is overridden by messages that have specialized JSON encodings.